### PR TITLE
Create the suggestion feed screen

### PR DESCRIPTION
**Pull request by annhit29**
In this pull request, 

Firstly, I test three composables SuggestionFeedContent, Suggestion and SuggestionBottomBar, where both SuggestionFeedContent and SuggestionBottomBar are newly created, and Suggestion is updated in order to have the visual part of the suggestion feed screen. 

- The composable SuggestionFeedContent is tested with both when the suggestion list is empty and when there are suggestions in the list of suggestions, to ensure these suggestions are well displayed if there are.
- The composable Suggestion is tested to ensure that the suggestion feed screen is correctly displayed.
- The composable SuggestionBottomBar is tested to ensure that the button for creating a suggestion exists and is displayed on the suggestions feed screen.

The suggestions feed screen is now visually correct, as expected.

Secondly, I test the composables SuggestionDetailPopup (to have the visual part of the suggestion details popup of a suggestion item) and SuggestionFeedContent, where both are newly created.
- The composable SuggestionDetailPopup is tested to ensure that the the suggestion details popup of a suggestion item displays correctly when comments are either present or absent, and when the address and website are either present or absent.
- The composable SuggestionFeedContent is tested to ensure that the suggestion items are clickable, the popup of the clicked suggestion item is displayed, and the overlay effect is working correctly.

Also, run the CI with the SuggestionFeedTest and SuggestionPopupTest classes, with dummy data so far for testing purposes.

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/SuggestionsViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import androidx.lifecycle.ViewModel
+import com.github.se.wanderpals.model.data.Suggestion
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+// todo: William has created this file, use his code
+
+open class SuggestionsViewModel(/*private val suggestionRepository: TripsRepository*/ ) :
+    ViewModel() {
+  // State flow to hold the list of trips
```
Comment (AgeX21) [inline app/src/main/java/com/github/se/wanderpals/model/viewmodel/SuggestionsViewModel.kt:16]: rename trips to suggestions

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionFeedContent.kt [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+          budget = 1500.0, // Entrance fee and other possible expenses
+          description =
+              "The Umeda Sky Building is a spectacular high rise building in the Kita district of Osaka, featuring a futuristic observatory, the Floating Garden.",
+          geoCords = GeoCords(latitude = 34.705938, longitude = 135.490357),
+          website = "http://www.kuchu-teien.com/",
+          imageUrl = "")
+
+//  /*
+      //these data are used for checking the UI, will be replaced with real data with William
+      val suggestionList = listOf(
```
Comment (AgeX21) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionFeedContent.kt:105]: careful with variable with same name, had trouble getting it to work locally

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionItem.kt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+              style = MaterialTheme.typography.bodyMedium,
+              fontSize = 14.sp)
+
+          Spacer(
+              modifier =
+                  Modifier.height(8.dp)) // Add spacing between the second row and the third row
+
+          // the third row
+          Text(
+              text = suggestion.text,
```
Comment (AgeX21) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionItem.kt:91]: Should be suggestion.stop.description (to be discussed with the team)

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Suggestion.kt [hunk 4]
```diff
@@ -1,12 +1,54 @@
...
+import com.github.se.wanderpals.ui.screens.suggestion.SuggestionTopBar
 
-/** The Finance screen. */
+/** The Suggestion screen. */
 @Composable
-fun Suggestion(tripId: String) {
-  Text(modifier = Modifier.testTag("suggestionScreen"), text = "Suggestion for trip $tripId")
+fun Suggestion(/*oldNavActions: NavigationActions,*/ tripId: String) {
+  val navController = rememberNavController()
+  val navActions = NavigationActions(navController)
```
Comment (AgeX21) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Suggestion.kt:24]: Should have a NavHost afterward. It crashes for now when pressing the button

**Review by AgeX21 — CHANGES_REQUESTED** [review]
Should add NavHost and working tests. Rest looks good to me !

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionFeedContent.kt (hunk 2 above)
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionFeedContent.kt]: Changed the variable name to `_suggestionList`, it's been working locally

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/SuggestionsViewModel.kt (hunk 1 above)
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/model/viewmodel/SuggestionsViewModel.kt]: Done

**Comment by annhit29** [comment]
You can check the code again, and all the tests are working now :D

**Review by AgeX21 — APPROVED** [review]
Looks good to me ! The NavHost is still missing but since it depends on my part, i'll add it when my branch gets on the main.

**Pull request merged**